### PR TITLE
Fixup file.line docs to be more clear and consistent

### PR DESCRIPTION
**Pull request by rallytime**
Refs saltstack/salt#31081

### CODE REVIEW

File: salt/modules/file.py [hunk 1]
```diff
@@ -1411,45 +1411,62 @@ def line(path, content, match=None, mode=None, location=None,
     '''
     .. versionadded:: 2015.8.0
 
-    Edit a line in the configuration file.
+    Edit a line in the configuration file. The ``path`` and ``content``
+    arguments are required, as well as passing in one of the ``mode``
+    options.
 
-    :param path:
+    .. note::
```
Comment (JensRantil) [inline salt/modules/file.py]: Would it make sense to add this note after the `mode` argument documentation?

**Review by JensRantil — COMMENTED** [review]
Awesome! Would it make sense to also document the `file.line` state in this PR?

File: salt/modules/file.py (hunk 1 above)
Comment (rallytime) [inline salt/modules/file.py]: Yeah that probably makes sense. I wasn't sure if that should have been exposed up higher in the docs or not. I'll move it. 👍

**Comment by rallytime** [comment]
@JensRantil Thanks for the review! I have moved the note as you suggested for the mode docs and I also added the doc changes to the file.line state function as well. Good catch! 

Do you know if there are any other open issues about the state docs? I will look around, but I thought you might have already filed one that this might close up also. :)

**Comment by JensRantil** [comment]
> Do you know if there are any other open issues about the state docs?

Don't think I ever did, but I'm happy you also updated it because it's been equally fuzzy.

File: salt/states/file.py [hunk 2]
```diff
@@ -2653,43 +2653,57 @@ def line(name, content, match=None, mode=None, location=None,
...
+        locations are:
 
-        :end:
+        - start
+            Place the content at the beginning of the file.
+        - end
             Place the content at the end of the file.
 
-    :param before:
+    before
```
Comment (JensRantil) [inline salt/states/file.py]: Is this used for all modes?

File: salt/states/file.py [hunk 3]
```diff
@@ -2653,43 +2653,57 @@ def line(name, content, match=None, mode=None, location=None,
...
         Regular expression or an exact case-sensitive fragment of the string.
 
-    :param after:
+    after
```
Comment (JensRantil) [inline salt/states/file.py]: Is this used for all modes?

File: salt/modules/file.py [hunk 4]
```diff
@@ -1411,45 +1411,62 @@ def line(path, content, match=None, mode=None, location=None,
...
+        locations are:
 
-        :end:
+        - start
+            Place the content at the beginning of the file.
+        - end
             Place the content at the end of the file.
 
-    :param before:
+    before
```
Comment (JensRantil) [inline salt/modules/file.py]: Is this used for all modes?

File: salt/modules/file.py [hunk 5]
```diff
@@ -1411,45 +1411,62 @@ def line(path, content, match=None, mode=None, location=None,
...
         Regular expression or an exact case-sensitive fragment of the string.
 
-    :param after:
+    after
```
Comment (JensRantil) [inline salt/modules/file.py]: Is this used for all modes?

File: salt/modules/file.py [hunk 6]
```diff
@@ -1458,31 +1475,33 @@ def line(path, content, match=None, mode=None, location=None,
...
         Create a backup of the original file with the extension:
         "Year-Month-Day-Hour-Minutes-Seconds".
 
-    :param quiet:
+    quiet
         Do not raise any exceptions. E.g. ignore the fact that the file that is
         tried to be edited does not exist and nothing really happened.
 
-    :param indent:
+    indent
```
Comment (JensRantil) [inline salt/modules/file.py]: Which modes does this apply to?

File: salt/states/file.py [hunk 7]
```diff
@@ -2698,15 +2712,15 @@ def line(name, content, match=None, mode=None, location=None,
...
         Create a backup of the original file with the extension:
         "Year-Month-Day-Hour-Minutes-Seconds".
 
-    :param quiet:
+    quiet
         Do not raise any exceptions. E.g. ignore the fact that the file that is
         tried to be edited does not exist and nothing really happened.
 
-    :param indent:
+    indent
```
Comment (JensRantil) [inline salt/states/file.py]: Which modes does this apply to?

**Review by JensRantil — COMMENTED** [review]
Had some more questions that might need clarification.

File: salt/states/file.py (hunk 2 above)
Comment (rallytime) [inline salt/states/file.py]: Good question - it looks like those options only take effect when using either the `insert` or `ensure` modes. I'll add a note about that.

File: salt/states/file.py (hunk 3 above)
Comment (rallytime) [inline salt/states/file.py]: The modes for `after` are the same as the `before` modes. I'll add a note about this as well.

File: salt/modules/file.py (hunk 4 above)
Comment (rallytime) [inline salt/modules/file.py]: Same answer here as above: only the ensure and insert modes use this option.

**Comment by rallytime** [comment]
Thanks again for the good questions @JensRantil. I have added a couple of clarifying sentences to these options for both the module and state docs.

**Pull request merged**